### ISSUE TITLE
mv_populating_from_existing_data_during_node_stop_test failed: Table 'ks.tm_table_mv_4' already exists

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#4270]
scylla version: 0eddf19432d12e68503194852d3183c20e9cf964
scylla-dtest version: 7a15b8f248f2c6ae6d01edc915e379421084d86f

Seen in [dtest-release/40/testReport/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test](http://jenkins.cloudius-systems.com:8080/view/master/job/scylla-master/job/dtest-release/40/testReport/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test/):
```
Table 'ks.tm_table_mv_4' already exists
...
Stacktrace
  File "/usr/lib64/python2.7/unittest/case.py", line 367, in run
    testMethod()
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 584, in mv_populating_from_existing_data_during_node_stop_test
    self._mv_populating_from_existing_data_during_changes_test('stop node', nodes=4, rf=3, mvs=10, prefill=40000, fail=False)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 635, in _mv_populating_from_existing_data_during_changes_test
    run_in_parallel(proc_functions)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 846, in run_in_parallel
    results = [task.result() for task in tasks]
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/.dtest_venv/lib/python2.7/site-packages/concurrent/futures/_base.py", line 455, in result
    return self.__get_result()
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/.dtest_venv/lib/python2.7/site-packages/concurrent/futures/thread.py", line 63, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 508, in _create_mvs_by_one_column
    mv_pk_column={'names': [tm.column_names_list[i]]}, wait_for_view_built=wait_for_view_built)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 722, in create_materialized_view
    self.parent_table.session.execute(statement)
  File "cassandra/cluster.py", line 2141, in cassandra.cluster.Session.execute
    return self.execute_async(query, parameters, trace, custom_payload, timeout, execution_profile, paging_state).result()
  File "cassandra/cluster.py", line 4033, in cassandra.cluster.ResponseFuture.result
    raise self._final_exception
```

FWIW, from [node2.log](http://jenkins.cloudius-systems.com:8080/view/master/job/scylla-master/job/dtest-release/40/artifact/logs-release.2/1550972846587_materialized_views_test.TestMaterializedViews.mv_populating_from_existing_data_during_node_stop_test/node2.log) it seems like the MV creation happened concurrently to stopping.  I'm not sure if this is how the test supposed to work...
```
INFO  2019-02-24 03:47:23,219 [shard 0] migration_manager - Create new view: org.apache.cassandra.config.CFMetaData@0x6000014cf880[cfId=2144b230-37d6-11e9-b6d1-000000000001,ksName=ks,cfName=tm_table_mv_4,cfType=Standard,comparator=org.apache.cassandra.db.marshal.CompositeType(org.apache.cassandra.db.marshal.Int32Type,org.apache.cassandra.db.marshal.UTF8Type),comment=,readRepairChance=0,dcLocalReadRepairChance=0.1,gcGraceSeconds=864000,keyValidator=org.apache.cassandra.db.marshal.IntegerType,minCompactionThreshold=4,maxCompactionThreshold=32,columnMetadata=[ColumnDefinition{name=id, type=org.apache.cassandra.db.marshal.IntegerType, kind=PARTITION_KEY, componentIndex=0, droppedAt=-9223372036854775808}, ColumnDefinition{name=clmn_int4, type=org.apache.cassandra.db.marshal.Int32Type, kind=CLUSTERING_COLUMN, componentIndex=0, droppedAt=-9223372036854775808}],compactionStrategyClass=class org.apache.cassandra.db.compaction.SizeTieredCompactionStrategy,compactionStrategyOptions={enabled=true},compressionParameters={sstable_compression=org.apache.cassandra.io.compress.LZ4Compressor},bloomFilterFpChance=0.01,memtableFlushPeriod=0,caching={"keys":"ALL","rows_per_partition":"ALL"},defaultTimeToLive=0,minIndexInterval=128,maxIndexInterval=2048,speculativeRetry=99.0PERCENTILE,triggers=[],isDense=false,version=2144b231-37d6-11e9-b6d1-000000000001,droppedColumns={},collections={},indices={}, viewInfo=ViewInfo{baseTableId=00a122c0-37d6-11e9-936e-000000000001, baseTableName=tm_table, includeAllColumns=false, whereClause=id IS NOT null AND clmn_int4 IS NOT null}]
INFO  2019-02-24 03:47:23,219 [shard 0] view - Finished building view ks.tm_table_mv_0
INFO  2019-02-24 03:47:23,224 [shard 0] compaction - Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-rw6aCT/test/node2/data/system_schema/scylla_tables-5d912ff1f7593665b2c88042ab5103dd/mc-18-big-Data.db:level=0, ]. 10480 bytes to 5419 (~51% of original) in 129ms = 0.04MB/s. ~256 total partitions merged to 1.
INFO  2019-02-24 03:47:23,224 [shard 0] compaction - Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-rw6aCT/test/node2/data/system_schema/columns-24101c25a2ae3af787c1b40ee1aca33f/mc-18-big-Data.db:level=0, ]. 11559 bytes to 6071 (~52% of original) in 129ms = 0.04MB/s. ~256 total partitions merged to 1.
INFO  2019-02-24 03:47:23,229 [shard 0] compaction - Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-rw6aCT/test/node2/data/system_schema/tables-afddfb9dbc1e30688056eed6c302ba09/mc-18-big-Data.db:level=0, ]. 13984 bytes to 7157 (~51% of original) in 107ms = 0.06MB/s. ~256 total partitions merged to 1.
INFO  2019-02-24 03:47:23,319 [shard 0] compaction_manager - Asked to stop
INFO  2019-02-24 03:47:23,319 [shard 1] compaction_manager - Asked to stop
INFO  2019-02-24 03:47:23,319 [shard 1] compaction_manager - Stopped
INFO  2019-02-24 03:47:23,437 [shard 0] schema_tables - Altering ks.tm_table id=00a122c0-37d6-11e9-936e-000000000001 version=52cbba82-60dd-301a-a53d-efe3cf4d309a
INFO  2019-02-24 03:47:23,438 [shard 0] schema_tables - Creating ks.tm_table_mv_4 id=2144b230-37d6-11e9-b6d1-000000000001 version=d0d19e95-0ce3-3251-87e9-7c7940a768a8
INFO  2019-02-24 03:47:23,442 [shard 1] query_processor - Column definitions for ks.tm_table changed, invalidating related prepared statements
INFO  2019-02-24 03:47:23,458 [shard 0] query_processor - Column definitions for ks.tm_table changed, invalidating related prepared statements
INFO  2019-02-24 03:47:23,463 [shard 0] view - Building view ks.tm_table_mv_4, starting at token -5384833187204020333
INFO  2019-02-24 03:47:23,476 [shard 0] compaction - Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-rw6aCT/test/node2/data/system_schema/views-9786ac1cdd583201a7cdad556410c985/mc-14-big-Data.db:level=0, ]. 14545 bytes to 7702 (~52% of original) in 321ms = 0.02MB/s. ~256 total partitions merged to 1.
INFO  2019-02-24 03:47:23,476 [shard 0] compaction - Compacted 2 sstables to [/jenkins/workspace/scylla-master/dtest-release/scylla/.dtest/dtest-rw6aCT/test/node2/data/system_schema/keyspaces-abac5682dea631c5b535b3d6cffd0fb6/mc-22-big-Data.db:level=0, ]. 10808 bytes to 5558 (~51% of original) in 321ms = 0.02MB/s. ~256 total partitions merged to 1.
INFO  2019-02-24 03:47:23,476 [shard 0] compaction_manager - Stopped
```

**Comment by bhalevy** [comment]
@psarna please look into this. Could it be related to the recent changes you submitted?

**Comment by nyh** [comment]
Interesting. I don't see how it's related to @psarna 's latest changes, though.
It seems that you are right, the dying node starts to build the view. But how can it cause the "create materialized view" command to fail?

I didn't manage to understand what the _mv_populating_from_existing_data_during_changes_test and _create_mvs_by_one_column code does. @juliayakovlev is it possible they accidentally list the same table name, "tm_table_mv_4" twice, and we end up trying to create the same table twice?

**Comment by juliayakovlev** [comment]
@nyh 

It's very weird because of by log we can see that the statement 
```
CREATE MATERIALIZED VIEW ks.tm_table_mv_4 AS SELECT clmn_int4 FROM ks.tm_table WHERE id IS NOT NULL and clmn_int4 IS NOT NULL PRIMARY KEY (id, clmn_int4);
```
has been executed just ones. The statement is printed before executing. So if it could be called twice, it could be printed twice. But it's not.
And as we can see that in the same time  the node2 was asked to stop and the MV creation statement was sent to node2. So the connection has been refused.
```
cassandra.connection: DEBUG: Message pushed from server: <EventMessage(event_type=u'SCHEMA_CHANGE', trace_id=None, event_args={'keyspace': u'ks', 'change_type': u'CREATED', 'target_type': u'TABLE', u'table': u'tm_table_mv_4'}, stream_id=-1)>
cassandra.connection: DEBUG: Message pushed from server: <EventMessage(event_type=u'STATUS_CHANGE', trace_id=None, event_args={'change_type': u'DOWN', 'address': ('127.0.97.2', 9042)}, stream_id=-1)>
cassandra.pool: WARNING: Error attempting to reconnect to 127.0.97.2, scheduling retry in 2.0 seconds: [Errno 111] Tried connecting to [('127.0.97.2', 9042)]. Last error: Connection refused
cassandra.pool: DEBUG: Reconnection error details
Traceback (most recent call last):
  File "cassandra/pool.py", line 208, in cassandra.pool._ReconnectionHandler.run
    conn = self.try_reconnect()
  File "cassandra/pool.py", line 279, in cassandra.pool._HostReconnectionHandler.try_reconnect
    return self.connection_factory()
  File "cassandra/connection.py", line 332, in cassandra.connection.Connection.factory
    conn = cls(host, *args, **kwargs)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/.dtest_venv/lib/python2.7/site-packages/cassandra/io/asyncorereactor.py", line 341, in __init__
    self._connect_socket()
  File "cassandra/connection.py", line 371, in cassandra.connection.Connection._connect_socket
    raise socket.error(sockerr.errno, "Tried connecting to %s. Last error: %s" % ([a[4] for a in addresses], sockerr.strerror or sockerr))
error: [Errno 111] Tried connecting to [('127.0.97.2', 9042)]. Last error: Connection refused
dtest: DEBUG: materialized_views_test.TestMaterializedViews.mv_populating_from_existing_data_during_node_stop_testmv_populating_from_existing_data_during_node_stop_test - FINISH: stop node node2
```
Seems the cqlsh (session.execute) returned wrong error message.

**Comment by psarna** [comment]
I have no idea what can cause `Table 'ks.tm_table_mv_4' already exists` and I haven't been able to reproduce, but this test is also prone to failures because we're waiting for view building to finish after we stopped the node - it won't, because view building will not continue until we're able to send view updates (i.e. the node is started again or removed). So, I'll prepare pull request to remedy at least this subproblem: http://jenkins.cloudius-systems.com:8080/view/master/job/scylla-master/job/dtest-release/43/testReport/junit/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test_2/

**Comment by slivne** [comment]
dtest was merged - closing this issue - we can reopen if it happens again

**Comment by bhalevy** [comment]
Reopening.

Hit again in [dtest-release/116/testReport/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test](https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/116/testReport/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test/):
```
Stacktrace
  File "/usr/lib64/python2.7/unittest/case.py", line 367, in run
    testMethod()
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 584, in mv_populating_from_existing_data_during_node_stop_test
    self._mv_populating_from_existing_data_during_changes_test('stop node', nodes=4, rf=3, mvs=10, prefill=40000, fail=False)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 635, in _mv_populating_from_existing_data_during_changes_test
    run_in_parallel(proc_functions)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 846, in run_in_parallel
    results = [task.result() for task in tasks]
  File "/usr/lib/python2.7/site-packages/concurrent/futures/_base.py", line 455, in result
    return self.__get_result()
  File "/usr/lib/python2.7/site-packages/concurrent/futures/thread.py", line 63, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 508, in _create_mvs_by_one_column
    mv_pk_column={'names': [tm.column_names_list[i]]}, wait_for_view_built=wait_for_view_built)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 722, in create_materialized_view
    self.parent_table.session.execute(statement)
  File "cassandra/cluster.py", line 2141, in cassandra.cluster.Session.execute
    return self.execute_async(query, parameters, trace, custom_payload, timeout, execution_profile, paging_state).result()
  File "cassandra/cluster.py", line 4033, in cassandra.cluster.ResponseFuture.result
    raise self._final_exception
'Table \'ks.tm_table_mv_4\' already exists
```

**Comment by slivne** [comment]
@bhalevy - did we fix this

**Comment by bhalevy** [comment]
> @bhalevy - did we fix this

We never got to the bottom of this issue, so I can't point to any specific fix.
There were quite a few fixes to materialized_views_test.py, but nothing on this path.
That said, this hasn't been seen it for a while though.
The [latest](https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/158/testReport/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test/) failure of this test case looks like a different reason (#4600)

**Comment by eliransin** [comment]
Still failing with certain flakiness.

**Comment by psarna** [comment]
I suspect this to be a race in the test itself. Let me explain. Here's how materialized views are created for this test:
```python
proc_functions = [change_func, {'func': self._create_mvs_by_one_column, 'args': (tm, mvs)}]
run_in_parallel(proc_functions)
```
Inside, `self._create_mvs_by_one_column` calls `create_materialized_view`, which is a function of the `MaterializedViewManager` class. This manager has a reference to its `TableManager` class and creates the views in such a way:
1. Pick a name for the materialized view
2. Check if this name is already registered in the list of materialized views this table has. This list is kept in the instance o `TableManager`
3. If the name we picked for this materialized view is already on the list, pick another name and create the view
4. put the used name in the list

Now, the problem is that we launch multiple such functions in parallel. A race can occur:
1. Thread A: a new name for a materialized view is picked: `ks.tm_table_mv_4`
2. Thread A: a new name is not on a list, continue to create the view
3. Thread B: a new name for a materialized view is picked: `ks.tm_table_mv_4`
4. Thread B: a new name is not on a list, continue to create the view
5. Thread B: crash, because thread A already created the view (hence - `Table already exists error`), but it didn't put it on the list yet

So, I think the problem is that `create_materialized_view` is not thread-safe, which is by itself fine, **but**, it's also ran in parallel, which is not fine. I wasn't able to reproduce this issue locally, so I cannot test if it's indeed the problem, but the code suggests so.

The easiest fix would be to stop running the view creation in parallel. A more complicated fix would be to ensure that creating a view via `create_materialized_view` is thread-safe.

**Comment by psarna** [comment]
Oh wait, I misread this code, only a single instance of `_create_mvs_by_one_column` is ran in parallel... so my whole story about the race does not apply.

I still think it has to do with the way views are created in the tests (that they somehow try to create a view that is already created), but I have to dig more to figure out what actually happens.

**Comment by psarna** [comment]
I tried to look at fresh logs, but this test doesn't seem to fail with `Table already exists`at all... it sometimes fail with `View not built`, which is a totally separate issue (perhaps the view simply has not finished building *yet*). Did anybody see actual fresh logs with this error? I wanted to use them for debugging, but found nothing from recent runs.

**Comment by psarna** [comment]
I'm 99% sure it's exactly the same problem as https://github.com/scylladb/scylla/pull/5408 and this issue will get fixed as well, but I'm not marking it with `Fixes #`, because I wasn't able to reproduce this particular problem.

**Comment by psarna** [comment]
@bhalevy did the issue reproduce after merging #5408? If not then it would be a safe bet to close this one.

**Comment by bhalevy** [comment]
> @bhalevy did the issue reproduce after merging #5408? If not then it would be a safe bet to close this one.

No, we're still hitting #5459 but not #4382 nor this bug.

**Comment by bhalevy** [comment]
Seen again in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/394/testReport/materialized_views_test/TestMaterializedViews/mv_populating_from_existing_data_during_node_stop_test/
Reopening...

Scylla version 70d914ad5b35d9c6b93032427461d2dbac47561f
```
Error Message
Table 'ks.tm_table_mv_4' already exists
Stacktrace
Traceback (most recent call last):
  File "/usr/lib64/python3.7/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib64/python3.7/unittest/case.py", line 645, in run
    testMethod()
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 604, in mv_populating_from_existing_data_during_node_stop_test
    self._mv_populating_from_existing_data_during_changes_test('stop node', nodes=4, rf=3, mvs=10, prefill=40000)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 656, in _mv_populating_from_existing_data_during_changes_test
    run_in_parallel(proc_functions)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 871, in run_in_parallel
    results = [task.result() for task in tasks]
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 871, in <listcomp>
    results = [task.result() for task in tasks]
  File "/usr/lib64/python3.7/concurrent/futures/_base.py", line 428, in result
    return self.__get_result()
  File "/usr/lib64/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/lib64/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/materialized_views_test.py", line 524, in _create_mvs_by_one_column
    mv_pk_column={'names': [tm.column_names_list[i]]}, wait_for_view_built=wait_for_view_built)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/scylla_tools.py", line 747, in create_materialized_view
    self.parent_table.session.execute(statement)
  File "cassandra/cluster.py", line 2141, in cassandra.cluster.Session.execute
  File "cassandra/cluster.py", line 4033, in cassandra.cluster.ResponseFuture.result
cassandra.AlreadyExists: Table 'ks.tm_table_mv_4' already exists
```

I wonder if the bug could be in the Cassandra driver, not dealing well with retrying after node2 is killed.

The node logs look ok to me...

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/394/artifact/logs-release.2/1582022213246_materialized_views_test.TestMaterializedViews.mv_populating_from_existing_data_during_node_stop_test/node1.log
```
INFO  2020-02-18 10:36:50,261 [shard 0] schema_tables - Creating ks.tm_table_mv_4 id=921baed0-523a-11ea-8327-000000000000 version=c2829c36-05e7-3264-85cf-87bec139ede5
INFO  2020-02-18 10:36:50,277 [shard 1] query_processor - Column definitions for ks.tm_table changed, invalidating related prepared statements
INFO  2020-02-18 10:36:50,283 [shard 0] query_processor - Column definitions for ks.tm_table changed, invalidating related prepared statements
INFO  2020-02-18 10:36:50,289 [shard 0] gossip - InetAddress 127.0.19.2 is now DOWN, status = shutdown
INFO  2020-02-18 10:36:50,292 [shard 0] view - Building view ks.tm_table_mv_4, starting at token 668445644334021178
...
INFO  2020-02-18 10:36:51,210 [shard 0] view - Finished building view ks.tm_table_mv_4
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/394/artifact/logs-release.2/1582022213246_materialized_views_test.TestMaterializedViews.mv_populating_from_existing_data_during_node_stop_test/node2.log
```
INFO  2020-02-18 10:36:50,109 [shard 0] migration_manager - Create new view: org.apache.cassandra.config.CFMetaData@0x6000035c8700[cfId=921baed0-523a-11ea-8327-000000000000,ksName=ks,cfName=tm_table_mv_4,cfType=Standard,comparator=org.apache.cassandra.db.marshal.CompositeType(org.apache.cassandra.db.marshal.Int32Type,org.apache.cassandra.db.marshal.UTF8Type),comment=,readRepairChance=0,dcLocalReadRepairChance=0.1,gcGraceSeconds=864000,keyValidator=org.apache.cassandra.db.marshal.IntegerType,minCompactionThreshold=4,maxCompactionThreshold=32,columnMetadata=[ColumnDefinition{name=id, type=org.apache.cassandra.db.marshal.IntegerType, kind=PARTITION_KEY, componentIndex=0, droppedAt=-9223372036854775808}, ColumnDefinition{name=clmn_int4, type=org.apache.cassandra.db.marshal.Int32Type, kind=CLUSTERING_COLUMN, componentIndex=0, droppedAt=-9223372036854775808}],compactionStrategyClass=class org.apache.cassandra.db.compaction.SizeTieredCompactionStrategy,compactionStrategyOptions={enabled=true},compressionParameters={sstable_compression=org.apache.cassandra.io.compress.LZ4Compressor},bloomFilterFpChance=0.01,memtableFlushPeriod=0,caching={"keys":"ALL","rows_per_partition":"ALL"},cdc={},defaultTimeToLive=0,minIndexInterval=128,maxIndexInterval=2048,speculativeRetry=99.0PERCENTILE,triggers=[],isDense=false,version=921baed1-523a-11ea-8327-000000000000,droppedColumns={},collections={},indices={}, viewInfo=ViewInfo{baseTableId=75be0670-523a-11ea-8815-000000000000, baseTableName=tm_table, includeAllColumns=false, whereClause=id IS NOT null AND clmn_int4 IS NOT null}]
...
INFO  2020-02-18 10:36:50,227 [shard 0] init - Signal received; shutting down
...
INFO  2020-02-18 10:36:50,263 [shard 0] schema_tables - Creating ks.tm_table_mv_4 id=921baed0-523a-11ea-8327-000000000000 version=c2829c36-05e7-3264-85cf-87bec139ede5
...
INFO  2020-02-18 10:36:52,641 [shard 0] init - Scylla version 666.development-0.20200217.70d914ad5b shutdown complete.
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/394/artifact/logs-release.2/1582022213246_materialized_views_test.TestMaterializedViews.mv_populating_from_existing_data_during_node_stop_test/node3.log
```
INFO  2020-02-18 10:36:50,258 [shard 0] schema_tables - Creating ks.tm_table_mv_4 id=921baed0-523a-11ea-8327-000000000000 version=c2829c36-05e7-3264-85cf-87bec139ede5
...
INFO  2020-02-18 10:36:50,288 [shard 0] view - Building view ks.tm_table_mv_4, starting at token 761526918485693167
...
INFO  2020-02-18 10:36:51,201 [shard 0] view - Finished building view ks.tm_table_mv_4
INFO  2020-02-18 10:36:52,548 [shard 0] rpc - client 127.0.19.2:7000: fail to connect: Connection refused
```

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/394/artifact/logs-release.2/1582022213246_materialized_views_test.TestMaterializedViews.mv_populating_from_existing_data_during_node_stop_test/node4.log
```
INFO  2020-02-18 10:36:50,256 [shard 0] schema_tables - Creating ks.tm_table_mv_4 id=921baed0-523a-11ea-8327-000000000000 version=c2829c36-05e7-3264-85cf-87bec139ede5
...
INFO  2020-02-18 10:36:50,281 [shard 0] view - Building view ks.tm_table_mv_4, starting at token -8727464244392362590
...
INFO  2020-02-18 10:36:51,127 [shard 0] view - Finished building view ks.tm_table_mv_4
INFO  2020-02-18 10:36:52,600 [shard 0] rpc - client 127.0.19.2:7000: fail to connect: Connection refused
```

**Comment by psarna** [comment]
Let's see if this failure is still observed after dcfdd917e got merged - it addresses a race in view building bookkeeping.

**Comment by slivne** [comment]
Closing this issue - @bhalevy if we hit this again please reopen / create new issue
